rhino-core/src/main/java/io/ryos/rhino/sdk/users/provider/FileBasedUserProviderImpl.java

```diff
@@ -16,33 +16,29 @@
 
 package io.ryos.rhino.sdk.users.provider;
 
+import io.ryos.rhino.sdk.io.FileResource;
 import io.ryos.rhino.sdk.users.CSVUserParserImpl;
 import io.ryos.rhino.sdk.users.UserParser;
-import io.ryos.rhino.sdk.users.VaultUserParserImpl;
 import io.ryos.rhino.sdk.users.data.User;
 import java.util.List;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 
 /**
- * Classpath file implementation of {@link UserProvider}.
+ * File based implementation of {@link UserProvider}.
+ * <p>
  *
  * @author Erhan Bagdemir
  */
-public class ClasspathUserProviderImpl implements UserProvider {
-
-  private static final Logger LOG = LogManager.getLogger(ClasspathUserProviderImpl.class);
-
+public class FileBasedUserProviderImpl implements UserProvider {
   private final String pathToFile;
   private final UserParser parser = new CSVUserParserImpl();
 
-  public ClasspathUserProviderImpl(final String pathToCSVFile) {
+  public FileBasedUserProviderImpl(final String pathToCSVFile) {
     this.pathToFile = pathToCSVFile;
   }
 
   @Override
   public List<User> getUsers() {
-    List<User> userList = parser.unmarshall(getClass().getResourceAsStream(pathToFile));
+    var userList = parser.unmarshall(new FileResource(pathToFile).getInputStream());
```

Comment:
If the file does not exists in the class path or in the file system, the user will see a NPE or something like that. Would prefer to check the path and inform the user that nothing was found if it does not exist.

Reply:
I have just renamed FileResource into ConfigResource. It sounds more descriptive to me.
Moreover, if the ConfigResource is not found, we throw a ConfigNotFoundException, instead of returning a null-value.